Editor/UI/GeneratedAssetsEditor.cs

```diff
@@ -270,6 +270,9 @@ private void ProcessSingleAsset(string directory, Object next)
                         $"Desired root {info.Root.Asset.name} for asset {next.name} is not a root asset");
                 }
 
+                // This is to keep project view clean
+                next.hideFlags |= HideFlags.HideInHierarchy | HideFlags.HideInInspector;
```

Comment:
It's useful to be able to select certain top-level assets (e.g. textures, animator controllers) for debugging purposes; we should limit this to certain specific types.

We'll also need to make sure that when we unpack these (https://github.com/bdunderscore/ndmf/blob/main/Editor/UI/GeneratedAssetsEditor.cs#L154) we unhide top-level assets.

Reply:
I thought it was already filtered to types that should be visible (i.e. top-level assets), and this change only apply hide flags to sub assets. Will it have cases that will include textures/animator controllers or other should-be-visible assets under other top-level assets?

Reply:
When you do a manual build we pack everything into a single asset file currently. This includes generated meshes, textures, animator controllers, expressions menu/parameters assets, etc...

For those assets that normally would be a top-level asset, it's useful to be able to select them for debugging purposes, so I'd prefer them be available (though, it could be an option, I suppose?).

After unpacking, these assets will also retain their hide flags, so we need to make certain that the unpacked assets are visible.

Reply:
So, if I don't misunderstand, the case I screenshotted in this PR should be the issue at where those assets generate (or combined), which make those states, transitions and state machine behaviors became visible?

Reply:
The case you screenshotted is where internal types like `AnimatorStateMachine`, `AnimatorState`, etc become visible. Those I agree should be hidden. The issue is we can't just hide all assets being added, as it's very useful to be able to access "top level" assets that are embedded in the combined asset file.

Reply:
Got it. I will setup a filter for these internal types.